electrum/transaction.py

```diff
@@ -1844,24 +1854,42 @@ def get_fee(self) -> Optional[int]:
         except MissingTxInputAmount:
             return None
 
+    def is_flag_valid(self, sighash) -> bool:
+        for flag in Sighash:
+            for base_flag in BaseSighash:
```

Comment:
For this single usage, I don't think it's worth it to define `BaseSighash`.
You could just list the types directly here: `[Sighash.SIGHASH_ALL, Sighash.SIGHASH_NONE, Sighash.SIGHASH_SINGLE]`.